L1Trigger/Phase2L1ParticleFlow/plugins/L1TPFProducer.cc

```diff
@@ -64,10 +64,11 @@ class L1TPFProducer : public edm::stream::EDProducer<> {
 
         edm::EDGetTokenT<math::XYZPointF> TokGenOrigin_;
 
-        // Region dump/coe
+        // Region dump/coe/apx
         FILE *fRegionDump;
-        l1tpf_impl::COEFile *fRegionCOE;
-        unsigned int neventscoemax, neventsproduced;
+        l1tpf_impl::COEPatternFile *fRegionCOE;
+        l1tpf_impl::APxPatternFile *fRegionAPx;
```

Comment:
are both needed at the same time?
if not, perhaps one could abstract the common interface in the base class, and have only one here (specificying the format in another parameter). 
that would allow duplicating all the lines for APx & COE later.

Reply:
I often need to produce both an APxPatternFile and a COEPatternFile using the same configuration. I would rather not run the code twice in order to produce these files. Also, all of the shared code is already in the PatternFile base class. Only the APx or COE specific code is contained in the two derived classes.

Reply:
>  If you still don't think this code should be integrated in the 11_1_X branch, where should it reside?

I don't have objections at integrating the code, but elements unrelated to L1PF should be put in more appropriate packages rather than dumping the whole thing into the Phase2L1ParticleFlow package.

That said, the admin of this repository is @rekovic